File: backend/src/main/java/vn/nguongocso/event/enums/ChainEventType.java
package vn.nguongocso.event.enums;

/**
 * Danh sách các loại sự kiện trong vòng đời chuỗi cung ứng.
 * @author Triệu Văn Đại
 */

public enum ChainEventType {

     //Sự kiện thu hoạch
    HARVEST,

     //Sự kiện đóng gói - Sản phẩm được đóng gói và dán nhãn.
    PACKAGING,

     //Sự kiện vận chuyển - Sản phẩm được di chuyển giữa các địa điểm.
    TRANSPORT,

     //Sự kiện thu mua - Sản phẩm được mua hoặc tiếp nhận.
    PROCUREMENT,

    //Sự kiện sửa lỗi - Điều chỉnh hoặc sửa dữ liệu sự kiện trước đó.
    CORRECTION
}

File: frontend/src/enums/chainEventType.ts
// Dùng object thay cho enum để tương thích với erasableSyntaxOnly
export const ChainEventType = {
  HARVEST: 'HARVEST',
  PACKAGING: 'PACKAGING',
  TRANSPORT: 'TRANSPORT',
  PROCUREMENT: 'PROCUREMENT',
  CORRECTION: 'CORRECTION',
} as const;

export type ChainEventType = (typeof ChainEventType)[keyof typeof ChainEventType];

// Nhãn hiển thị tiếng Việt
export const ChainEventTypeLabel: Record<ChainEventType, string> = {
  [ChainEventType.HARVEST]: 'Thu hoạch',
  [ChainEventType.PACKAGING]: 'Đóng gói',
  [ChainEventType.TRANSPORT]: 'Vận chuyển',
  [ChainEventType.PROCUREMENT]: 'Thu mua',
  [ChainEventType.CORRECTION]: 'Đính chính',
};